molecule_hetznercloud/playbooks/create.yml

```diff
@@ -5,153 +5,121 @@
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
-        state: "present"
-      loop: "{{ molecule_yml.platforms|molecule_get_hetznercloud_networks('subnetworks') }}"
+    - name: Attach server to subnetwork(s) # noqa jinja[invalid]
+      hetzner.hcloud.hcloud_server_network:
+        network: "{{ resource_namespace }}-{{ item.network_name }}"
+        server: "{{ resource_namespace }}-{{ item.server_name }}"
+        ip: "{{ item.ip | ansible.utils.ipaddr('address') }}"
+        state: present
+      loop: "{{ molecule_yml.platforms | molecule_get_hetznercloud_subnetworks() }}"
 
-    - name: Populate instance config dict
-      set_fact:
-        instance_conf_dict:
-          {
-            "instance": "{{ item.hcloud_server.name }}",
-            "ssh_key_name": "{{ ssh_key_name }}",
-            "address": "{{ item.hcloud_server.ipv4_address }}",
-            "user": "{{ ssh_user }}",
-            "port": "{{ ssh_port }}",
-            "identity_file": "{{ ssh_path }}",
-            "volumes": "{{ item.item.item.volumes | default({}) }}",
-            "networks": "{{ item.item.item.networks | default({}) | dict2items(key_name='name', value_name='data') }}",
-          }
-      with_items: "{{ hetzner_jobs.results }}"
-      register: instance_config_dict
-      when: server.changed | bool
+    - name: Create instance config
+      block:
+        - name: Populate instance config dict
+          ansible.builtin.set_fact:
+            instance_conf_dict:
+              {
+                "instance": "{{ item.item.item.name }}",
+                "address": "{{ item.hcloud_server.ipv4_address }}",
+                "user": "{{ ssh_user }}",
+                "port": "{{ ssh_port }}",
+                "ssh_key_name": "{{ ssh_key_name }}",
+                "identity_file": "{{ ssh_key_path }}",
+                "volumes": "{{ item.item.item.volumes | default([]) }}",
+                "networks": "{{ item.item.item.networks | default({}) | dict2items(key_name='name') }}",
+              }
+          loop: "{{ servers.results }}"
+          register: instance_config_dict
 
-    - name: Convert instance config dict to a list
-      set_fact:
-        instance_conf: "{{ instance_config_dict.results | map(attribute='ansible_facts.instance_conf_dict') | list }}"
-      when: server.changed | bool
+        - name: Convert instance config dict to a list
+          ansible.builtin.set_fact:
+            instance_conf: "{{ instance_config_dict.results | map(attribute='ansible_facts.instance_conf_dict') | list }}"
 
-    - name: Dump instance config
-      copy:
-        content: |
-          # Molecule managed
+        - name: Dump instance config
+          ansible.builtin.copy:
+            content: |
+              # Molecule managed
 
-          {{ instance_conf | to_json | from_json | to_yaml }}
-        dest: "{{ molecule_instance_config }}"
-      when: server.changed | bool
+              {{ instance_conf | to_json | from_json | to_yaml }}
```

Comment:
Please add a comment to explain why `to_json | from_json | to_yaml` is needed (vs. just `| to_yaml`).

Reply:
The origin of this workaround is here: https://github.com/ansible/ansible/issues/20885

Since we only support ansible-core >=2.13 I believe we can drop the workaround.